vendor/forms.py

```diff
@@ -58,6 +59,7 @@ def __init__(self, *args, **kwargs):
         self.fields['end_date'].widget.attrs['class'] = 'datepicker'
         self.fields['term_start_date'].widget.attrs['class'] = 'datepicker'
         self.fields['available'].label = _('Available to Purchase')
+        self.fields['products'].queryset = Product.objects.filter(site=Site.objects.get_current(), available=True)
```

Comment:
Site.objects.get_current() is dependent on the SITE_ID to work.

Reply:
Changed to:
`self.fields['products'].queryset = Product.objects.filter(site=request.site, available=True)`
 and added the Mixin to pass the request with the site set in the View.:
 
 ```

def get_site_from_request(request):
    if hasattr(request, 'site'):
        return request.site
    return get_current_site(request)

class PassRequestToFormKwargsMixin:

    def get_form_kwargs(self):
        kwargs = super().get_form_kwargs()
        self.request.site = get_site_from_request(self.request)
        kwargs['request'] = self.request
        return kwargs
```

Reply:
This is the line where the problem is:
`self.request.site = get_site_from_request(self.request)`

this line modifies the request object from within the Mixin.  Requests should only be modified by middle wear.